### PR TITLE
MAINT-27009: Add checks on user profile fields.

### PR DESCRIPTION
**Pull request by mdiha**
This PR regroups some fixes to add different input checks to fields in edit User profile.

### CODE REVIEW

File: webapp/portlet/src/main/resources/locale/portlet/social/ProfileWorkExperience_en.properties [hunk 1]
```diff
@@ -11,6 +11,7 @@ profileWorkExperiences.endDate=End Date
 profileWorkExperiences.stillInPosition=Still in position
 profileWorkExperiences.removeExperience=Remove Experience
 profileWorkExperiences.formValidationError=Please check validation information in fields.
+profileWorkExperiences.invalidOrganizationName=The length of the text field "Organization" must be between "0" and "250".
```
Comment (ghost) [inline webapp/portlet/src/main/resources/locale/portlet/social/ProfileWorkExperience_en.properties]: Please avoid using quotes and static values in resource bundles

**Review by ghost — DISMISSED** [review]

File: webapp/portlet/src/main/resources/locale/portlet/social/ProfileWorkExperience_en.properties (hunk 1 above)
Comment (mdiha) [reply webapp/portlet/src/main/resources/locale/portlet/social/ProfileWorkExperience_en.properties]: I am already intending to update it.
PS: the PR is not finished yet.

File: webapp/portlet/src/main/resources/locale/portlet/social/ProfileContactInformation_en.properties [hunk 2]
```diff
@@ -25,3 +25,5 @@ profileContactInformation.team=Team
 profileContactInformation.profession=Profession
 profileContactInformation.country=Country
 profileContactInformation.city=City
+profileContactInformation.error.invalidField=Only alpha, digit, dash and underscore characters allowed for the field "{0}"
```
Comment (ahamdi) [inline webapp/portlet/src/main/resources/locale/portlet/social/ProfileContactInformation_en.properties]: Please avoid using double quotes

File: webapp/portlet/src/main/webapp/profile-contact-information/components/ProfileContactInformationDrawer.vue [hunk 3]
```diff
@@ -215,6 +215,84 @@ export default {
   methods: {
     save() {
       this.error = null;
+      const regex =  /^[a-zA-Z0-9-_]+$/;
+      const startWithRegex =  /^\d/;
+      const isValidFirstName = regex.test(this.userToSave.firstname);
+      const isValidLastName = regex.test(this.userToSave.lastname);
+
```
Comment (ahamdi) [inline webapp/portlet/src/main/webapp/profile-contact-information/components/ProfileContactInformationDrawer.vue]: firstname, lastname validation is based on custom validators as described in https://docs.exoplatform.org/en/5.3/Configuration.html?highlight=validator#custom-data-validators-configuration
We need to load those configuration to check each field validity.

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: webapp/portlet/src/main/resources/locale/portlet/social/ProfileContactInformation_en.properties (hunk 2 above)
Comment (mdiha) [reply webapp/portlet/src/main/resources/locale/portlet/social/ProfileContactInformation_en.properties]: Okay but that is how the message is set.

File: webapp/portlet/src/main/webapp/profile-contact-information/components/ProfileContactInformationDrawer.vue (hunk 3 above)
Comment (mdiha) [reply webapp/portlet/src/main/webapp/profile-contact-information/components/ProfileContactInformationDrawer.vue]: The constants "isValidFirstName" and "isValidLastName" aren't the final checks they are just checks for the validity of no special characters, custom validators are checked in the following. Maybe i should rename them. WDYT ?

Comment (ahamdi) [reply webapp/portlet/src/main/webapp/profile-contact-information/components/ProfileContactInformationDrawer.vue]: The regex itself could be configured by customers to enable or disable charcters, field max and min length etc ...

Comment (mdiha) [reply webapp/portlet/src/main/webapp/profile-contact-information/components/ProfileContactInformationDrawer.vue]: Done

**Review by ahamdi — APPROVED** [review]

**Pull request merged**